fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -104,136 +104,155 @@ public boolean visit(String elementName, int elementIndex, Visitable visitable)
 
     @Override
     public boolean visit(java.lang.String elementName, int elementIndex, com.ibm.fhir.model.type.Boolean _boolean) {
-        Parameter p = new Parameter();
-        if (!TOKEN.equals(searchParamType)) {
-            throw invalidComboException(searchParamType, _boolean);
-        }
-        p.setName(searchParamCode);
-        p.setValueSystem("http://terminology.hl7.org/CodeSystem/special-values");
-        if (_boolean.getValue() != null && _boolean.getValue()) {
-            p.setValueCode("true");
-        } else {
-            p.setValueCode("false");
+        if (_boolean.hasValue()) {
```

Comment:
can _boolean be null?

Reply:
technically it could if someone invoked the method that way, but it will never be null from FHIRPersistenceJDBCImpl.extractSearchParameters because its invoked via double dispatch and the datatype visit methods have null checks in AbstractVisitable:
```
protected void accept(java.lang.String value, java.lang.String elementName, Visitor visitor) {
        if (value != null) {
            visitor.visit(elementName, value);
        }
    }
```